pkg/uploader/kopia/shim.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright The Velero Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kopia
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/content"
+	"github.com/kopia/kopia/repo/manifest"
+	"github.com/kopia/kopia/repo/object"
+)
+
+type shimRepository struct {
+	udmRepo udmrepo.BackupRepo
+}
+
+type shimObjectWriter struct {
+	repoWriter udmrepo.ObjectWriter
+}
+
+type shimObjectReader struct {
+	repoReader udmrepo.ObjectReader
+}
+
+func NewShimRepo(repo udmrepo.BackupRepo) repo.RepositoryWriter {
+	return &shimRepository{
+		udmRepo: repo,
+	}
+}
+
+func (sr *shimRepository) OpenObject(ctx context.Context, id object.ID) (object.Reader, error) {
+	reader, err := sr.udmRepo.OpenObject(ctx, udmrepo.ID(id))
+	if reader == nil {
+		return nil, err
+	}
+
+	return &shimObjectReader{
+		repoReader: reader,
+	}, err
+}
+
+func (sr *shimRepository) VerifyObject(ctx context.Context, id object.ID) ([]content.ID, error) {
```

Comment:
Return something like "error not supported", the unified repo doesn't support this method

Reply:
Already modified it